### ISSUE TITLE
Should UnusedFunctionArgument ignore _ arguments?

### ISSUE THREAD

**Issue by non-Jedi** [julia-vscode/StaticLint.jl#163]
I have an iterable mutable type which internally manages its state instead of using the second argument to `Base.iterate`. But to make it an iterable I still have to define the full `Base.iterate` interface so I do:

```julia
function Base.iterate(x::MyType, _=nothing)
```

StaticLint correctly flags the `_` as being unused in the function body, but I'm already aware that it's unused; that's why I've named it `_` which cannot be used as an "rvalue".

An example of this pattern but without using `_` from Base is the `iterate` method for `Base.EachLine` which would get flagged. In general for a language like Julia which has ad-hoc interfaces like this, I'm not sure this lint pass makes sense.

**Comment by ZacLN** [comment]
Hi, I was conscious of this being a more general issue (i.e. it seems fairly likely to occur in a multiple dispatch setting without indicating an issue) which is why I hesitated to merge it at first. To account for this on the vscode side of things this error gets reported as a DiagnosticSeverity.Hint with the Unnecessary DiagnosticTag meaning this gets visualised as the identifier being faded out. 

It seems easier to handle/assess the significance of an error/hint on the client side but if there was a particular need we could associate the `StaticLint.LintCodes` with a severity rating (I'd likely the same ratings as in the language server protocol)

**Comment by dalejordan** [comment]
I think handling it on the client side is problematic.  For emacs the client is eglot which services a number of languages.  eglot-jl.jl is a very thin wrapper which just tells eglot how to start the LS.  Maybe allowing some customization of linter decisions would be useful.  I use "_" as a nameless parameter often.

**Comment by non-Jedi** [comment]
@dalejordan You can already disable this lint through configuration with per-directory local variables. See the eglot README. It's "julia.lint.useoffuncargs".

**Comment by davidanthoff** [comment]
Couldn't one just write `Base.iterate(x::MyType, ::Any=nothing)` or `Base.iterate(x::MyType, ::Nothing=nothing)` instead? Isn't that the idiomatic Julia way to signal that one doesn't want to use the argument in the function?

**Comment by non-Jedi** [comment]
I wasn't actually aware that you could assign a default argument without having a variable binding. Still looks funky to me, but that certainly covers this case imo.
